nfadeeva/assignment3/dictionary.py

```diff
@@ -0,0 +1,61 @@
+import collections
+from typing import List
+
+
+class Node:
+    def __init__(self):
+        self.children = collections.defaultdict(Node)
+        self.is_word = False
+
+
+class Dictionary:
+    """
+    Dictionary class that looks like a trie
+    """
+
+    def __init__(self, data: List[str]):
+        """
+        :param data: list of words
+        :type data: List[str]
+        """
+
+        self.root = Node()
+        for word in data:
+            self.put(word)
+
+    def put(self, word):
```

Comment:
nit: use a different name. Conventionally, a method called "put" usually decreases a reference count on an object, it means that some user is no longer using that object. Here you are doing exactly the opposit.
You could call the method e.g. "add".